### PR TITLE
fix and enable wasm on xplat

### LINKED ISSUE #3561: Enable WebAssembly on xplat
- [ ] Remove `exclude_xplat` tag from test folder `WasmSpec` and from individual tests in test folder `wasm`
- [ ] Fix failing tests
- [ ] Enable WebAssembly in release builds (remove `#if defined(_WIN32) || (defined(__clang__) && defined(ENABLE_DEBUG_CONFIG_OPTIONS))` in CommonDefines.h

### PR DESCRIPTION
**Pull request by MikeHolman**
fix issues with calling convention: we are using stack based calling convention, so we can cleanup a lot of code that was complicated by xmm registers.

fix issue where JIT was calling wrong method for floorf and ceilf

Resolves #3561

### CODE REVIEW

File: test/wasm/signextend.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
 /* global assert,testRunner */ // eslint rule
 WScript.Flag("-WasmSignExtends");
 WScript.Flag("-WasmI64");
-WScript.LoadScriptFile("../UnitTestFrameWork/UnitTestFrameWork.js");
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
```
Comment (obastemur) [inline test/wasm/signextend.js]: why this change?

Comment (Cellule) [reply test/wasm/signextend.js]: The change for the casing is necessary, however, we should still use forward slashes

File: test/wasm/binary.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
 //-------------------------------------------------------------------------------------------------------
 
 /* global assert,testRunner */ // eslint rule
-WScript.LoadScriptFile("../UnitTestFrameWork/UnitTestFrameWork.js");
+WScript.LoadScriptFile("../UnitTestFramework/UnitTestFramework.js");
 WScript.LoadScriptFile("../wasmspec/testsuite/harness/wasm-constants.js");
```
Comment (obastemur) [inline test/wasm/binary.js]: Instead of `wasmspec` use `WasmSpec` or rename the folder.

**Comment by obastemur** [comment]
I'm happy that we are about to enable wasm on xplat! @MikeHolman @Cellule Thanks for this!

couple of small comments

LGTM!

**Comment by Cellule** [comment]
:lgtm:
Nice fixes! 
The calling convention looks so much simpler now :)

---

Reviewed 9 of 11 files at r1, 7 of 9 files at r2, 1 of 2 files at r3, 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/microsoft/chakracore/4531#-:-L2bDcFghkykcc4gIRHU:bq6qfs6)*

File: test/AsmJs/rlexe.xml [hunk 3]
```diff
@@ -522,6 +524,8 @@
     <default>
       <files>MathBuiltinsCall.js</files>
       <baseline>MathBuiltinsCall.baseline</baseline>
+      <!-- mac gives has some test which gives different precision result, so exclude it -->
```
Comment (obastemur) [inline test/AsmJs/rlexe.xml]: nit: open an issue to track these?

Comment (MikeHolman) [reply test/AsmJs/rlexe.xml]: I can, but do you think it's worth tracking? this is difference in precision with result of atan, which spec says is implementation defined approximation.

Comment (obastemur) [reply test/AsmJs/rlexe.xml]: I see. I thought only some tests have difference but others don't. If this is not the case, well, you are right, no point tracking it!

Generally speaking;
What we did with some of the date time and number formatting testing was to enable-disable per platform. i.e. `CH` supports the line below that checks if the system running the test is a macOS.

```
var isMac = (WScript.Platform && WScript.Platform.OS == 'darwin');
```

**Comment by MikeHolman** [comment]
@dotnet-bot test OSX static_osx_osx_debug

**Pull request merged**